source/framework/core/src/TRestAnalysisTree.cxx

```diff
@@ -982,6 +1004,88 @@ Int_t TRestAnalysisTree::WriteAsTTree(const char* name, Int_t option, Int_t bufs
     return result;
 }
 
+/// <summary>
+/// Add a series output file like TChain.
+/// </summary>
+/// <param name="file"> The input file that contains another AnalysisTree with same run id </param>
+/// <returns></returns>
+Bool_t TRestAnalysisTree::AddChainFile(string _file) {
+    TFile* file = new TFile(_file.c_str(), "read");
+    if (!file->IsOpen()) {
+        warning << "TRestAnalysisTree::AddChainFile(): failed to open file " << _file << endl;
+        return false;
+    }
+
+    TRestAnalysisTree* tree = (TRestAnalysisTree*)file->Get("AnalysisTree");
+    if (tree != nullptr && tree->GetEntry(0)) {
+        if (tree->GetEntry(0) > 0) {
+            if (tree->GetRunOrigin() == this->GetRunOrigin()) {
+                // this is a valid tree
+                delete tree;
```

Comment:
I don't think is a good practise to `delete` an object which has not been created with `new`, you can fall in an unexpected behaviour.

Reply:
But this is what returned by ROOT. We can only trust that TFile::Get() returns the pointer which is deleteable.

Reply:
For sure you can delete the pointer, however `root` has his own memory management, which means that it can try to access de `tree` that you just delete, which can end up in a seg fault.

Reply:
I don't understand. In which case will ROOT access such object? Or there is any "safe" method provided by ROOT to delete it?

Reply:
The `tree` in fact belongs to the `TFile`, when the `TFile` is destroyed it migth try to `delete` the tree, so you can fall in a double free with an unexpected behaviour.
https://stackoverflow.com/questions/9169774/what-happens-in-a-double-delete

Reply:
I did a test and there is no such double free fault if I delete the tree manually. There is also no memory leak if I don't delete the tree. So I guess ROOT has some mechanism to prevent object from file being deleted. Either delete or not is OK.

Reply:
But I still need to delete the tree. Otherwise the initializaion of the TChain will be problematic.

Reply:
@juanangp Can you approve this PR? I believe those problems are trivial